doc/nrf/gs_testing.rst

```diff
@@ -120,3 +120,68 @@ To connect to the nRF9160-based kit with LTE Link Monitor, perform the following
    .. note::
 
       In the case of nRF9160 DK, the reset button must be pressed to restart the device and to start the application.
+
+.. _gs_debugging:
+
+Debugging an application
+************************
+
+To debug the application core firmware, you need a single debug session.
+Set up the debug session as described in `Debugging an application`_ in the |nRFVSC| documentation.
+
+To debug the firmware running on the network core, you need to set up two separate debug sessions: one for the network core and one for the application core.
+When debugging the network core, the application core debug session runs in the background and you can debug both cores if needed.
+
+Complete the following steps to start debugging the network core:
+
+1. Set up sessions for the application core and network core as mentioned in `Debugging an application`_.
+#. Select the appropriate CPU for debugging in each session, corresponding to the application core and the network core of your SoC, respectively.
+#. Once both sessions are established, execute the code on the application core.
+
+   The startup code releases the ``NETWORK.FORCEOFF`` signal to start the network core and allocates the necessary GPIO pins for it.
+#. Start code execution on the network core in the other debug session.
+
+If you want to reset the network core while debugging, make sure to first reset the application core and execute the code.
+
+.. note::
+  Debugging firmware on the application core in the non-secure domain (that is, when using the ``_ns`` build target), is currently not supported.
+
+Debugging tools
+===============
+
+For debugging with the |nRFVSC|, read `Debugging an application`_ in the |nRFVSC| documentation.
+After adding the required Kconfig options to the :file:`prj.conf` file, the recommended approach is to use the `GNU Project Debugger`_ (GDB tool) for debugging.
+
+If you are working from the command line, you can use west with the GDB tool.
+For details, read the :ref:`Debugging with west debug <zephyr:west-debugging>` section on the :ref:`zephyr:west-build-flash-debug` page in the Zephyr documentation.
+
+A useful tool for debugging mesh networking protocols, such as :ref:`ug_thread` or :ref:`ug_zigbee`, is the `nRF Sniffer for 802.15.4`_.
```

Comment:
btw. nRF Sniffer is also very useful for Bluetooth communication

Reply:
IF we remove GDB information from this section we can name it e.g. `Sniffing the communication`